contracts/infrastructure/WalletFactory.sol

```diff
@@ -222,4 +223,60 @@ contract WalletFactory is Owned, Managed {
         require(_guardian != (address(0)), "WF: guardian cannot be null");
         require(_version > 0, "WF: invalid _version");
     }
+
+    /**
+     * @notice Refunds the creation of the wallet when provided with a vild signature from the wallet owner.
+     * @param _wallet The wallet created
+     * @param _owner The owner address
+     * @param _refundAmount The amount to refund
+     * @param _refundToken The token to use for the refund
+     * @param _ownerSignature A signature from the wallet owner approving the refund amount and token. 
+     */
+    function validateAndRefund(
+        address _wallet,
+        address _owner,
+        uint256 _refundAmount,
+        address _refundToken,
+        bytes memory _ownerSignature
+    )
+        internal
+    {
+        bytes32 signedHash = keccak256(abi.encodePacked(
+                "\x19Ethereum Signed Message:\n32",
+                keccak256(abi.encodePacked(_refundAmount, _refundToken))
+            ));
+        address signer = Utils.recoverSigner(signedHash, _ownerSignature, 0);
+        if (signer == _owner) {
+            if (_refundToken == ETH_TOKEN) {
+				invokeWallet(_wallet, refundAddress, _refundAmount, "");
```

Comment:
Something with spacing here

Reply:
I don't understand, it's properly indented on my IDE... not the first time it happens.

Reply:
It's fixed.